src/az_ulib_ipc/az_ulib_ipc.c

```diff
@@ -0,0 +1,356 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+#include <stdint.h>
+
+#include "azure_macro_utils/macro_utils.h"
+#include "umock_c/umock_c_prod.h"
+
+#include "az_pal_os_api.h"
+#include "az_ulib_action_api.h"
+#include "az_ulib_base.h"
+#include "az_ulib_descriptor_api.h"
+#include "az_ulib_ipc_api.h"
+#include "internal/az_ulib_ipc.h"
+#include "ucontract.h"
+#include "ulib_config.h"
+#include "ulib_port.h"
+#include "ulib_result.h"
+#include "ulog.h"
+
+/*
+ * IPC is a singleton component, and shall be initialized only once.
+ */
+static _az_ulib_ipc *ipc = NULL;
+
+static _az_ulib_ipc_interface *get_interface(
+    const char *const name,
+    az_ulib_version version,
+    AZ_ULIB_VERSION_MATCH_CRITERIA match_criteria) {
+  _az_ulib_ipc_interface *result = NULL;
+
+  for (size_t i = 0; i < AZ_ULIB_CONFIG_MAX_IPC_INTERFACE; i++) {
+    if ((ipc->interface_list[i].interface_descriptor != NULL)
+        && (strcmp(ipc->interface_list[i].interface_descriptor->name, name) == 0)
```

Comment:
>strcmp [](start = 12, length = 6)

should we be looking to use the `strncmp` with some max interface name in the config?

Reply:
[https://github.com/Azure/azure-ulib-c/issues/42](https://github.com/Azure/azure-ulib-c/issues/42)

---
In reply to: [339134157](https://github.com/Azure/azure-ulib-c/pull/40#discussion_r339134157) [](ancestors = 339134157)